# <center> Jeodyshan Obra IS 362 Assignemnt 5 </center>

In [1]:
import pandas as pd
from pandas import DataFrame
df = pd.read_csv("airports.csv")

## Displaying airports data

In [2]:
pd.read_csv("airports.csv")

faa                                  name        lat         lon   alt  \
0     04G                     Lansdowne Airport  41.130472  -80.619583  1044   
1     06A         Moton Field Municipal Airport  32.460572  -85.680028   264   
2     06C                   Schaumburg Regional  41.989341  -88.101243   801   
3     06N                       Randall Airport  41.431912  -74.391561   523   
4     09J                 Jekyll Island Airport  31.074472  -81.427778    11   
5     0A9        Elizabethton Municipal Airport  36.371222  -82.173417  1593   
6     0G6               Williams County Airport  41.467306  -84.506778   730   
7     0G7         Finger Lakes Regional Airport  42.883565  -76.781232   492   
8     0P2          Shoestring Aviation Airfield  39.794824  -76.647191  1000   
9     0S9                 Jefferson County Intl  48.053809 -122.810644   108   
10    0W3                Harford County Airport  39.566838  -76.202403   409   
11    10C                    Galt Field Airport  42.402889  -88.375111   875   
12    17G  Port Bucyrus-Crawford County Airport  40.781556  -82.974806  1003   
13    19A                Jackson County Airport  34.175864  -83.561597   951   
14    1A3         Martin Campbell Field Airport  35.015806  -84.346833  1789   
15    1B9                   Mansfield Municipal  42.000133  -71.196771   122   
16    1C9                  Frazier Lake Airpark  54.013333 -124.768333   152   
17    1CS            Clow International Airport  41.695974  -88.129231   670   
18    1G3                    Kent State Airport  41.151389  -81.415111  1134   
19    1G4             Grand Canyon West Airport  35.899904 -113.815674  4813   
20    1H2            Effingham Memorial Airport  39.070000  -88.534000   585   
21    1OH                       Fortman Airport  40.555325  -84.386619   885   
22    1RL                 Point Roberts Airpark  48.979722 -123.078889    10   
23    23M                             Clarke CO  32.051700  -88.443400   320   
24    24C                   Lowell City Airport  42.953920  -85.343906   681   
25    24J               Suwannee County Airport  30.300125  -83.024694   104   
26    25D                   Forest Lake Airport  45.247746  -92.994385   925   
27    29D                    Grove City Airport  41.146028  -80.167750  1371   
28    2A0                    Mark Anton Airport  35.486250  -84.931083   718   
29    2B2                   Plum Island Airport  42.795361  -70.839444    11   
...   ...                                   ...        ...         ...   ...   
1428  X59                    Valkaria Municipal  27.960861  -80.558333    26   
1429  XFL                Flagler County Airport  29.282100  -81.121200    33   
1430  XNA                  NW Arkansas Regional  36.281869  -94.306811  1287   
1431  XZK            Amherst Amtrak Station AMM  42.375000  -72.511389   258   
1432  Y51                     Municipal Airport  43.579360  -90.896474  1292   
1433  Y72                          Bloyer Field  43.976222  -90.480611   966   
1434  YAK                               Yakutat  59.301200 -139.393700    33   
1435  YIP                            Willow Run  42.237928  -83.530408   716   
1436  YKM  Yakima Air Terminal McAllister Field  46.568200 -120.544000  1095   
1437  YKN                           Chan Gurney  42.871100  -97.396900  1200   
1438  YNG                Youngstown Warren Rgnl  41.260736  -80.679097  1196   
1439  YUM                   Yuma Mcas Yuma Intl  32.656578 -114.605980   216   
1440  Z84                                 Clear  64.301203 -149.120144   552   
1441  ZBP                          Penn Station  39.307222  -76.615556    66   
1442  ZFV          Philadelphia 30th St Station  39.955700  -75.182000     0   
1443  ZPH                     Municipal Airport  28.228056  -82.155833    90   
1444  ZRA           Atlantic City Rail Terminal  39.366500  -74.442000     8   
1445  ZRD                         Train Station  37.534300  -77.429450    26   
1446  

In [7]:
df.loc[(df['lat'] >= 71)]

faa                        name        lat         lon  alt  tz dst  \
230  BRW  Wiley Post Will Rogers Mem  71.285446 -156.766003   44  -9   A   
417  EEN     Dillant Hopkins Airport  72.270833   42.898333  149  -5   A   

                 tzone  
230  America/Anchorage  
417                NaN

### By filtering the data 71 latitude is the highest point of Alaska and Wiley Post WIll Rogers is the northernmost airport in America

In [12]:
df.loc[(df['lon'] >= -68)]

faa                           name        lat         lon   alt  tz dst  \
396   DVT  Deer Valley Municipal Airport  33.411700  112.457000  1478   8   A   
417   EEN        Dillant Hopkins Airport  72.270833   42.898333   149  -5   A   
444   EPM     Eastport Municipal Airport  44.910111  -67.012694    45  -5   A   
624   HUL                   Houlton Intl  46.123083  -67.792056   489  -5   A   
942   MYF               Montgomery Field  32.475900  117.759000    17   8   A   
1290  SYA                   Eareckson As  52.712275  174.113620    98  -9   A   

                  tzone  
396      Asia/Chongqing  
417                 NaN  
444    America/New_York  
624    America/New_York  
942      Asia/Chongqing  
1290  America/Anchorage

### By filtering the data -68 longitude is the easternmost part of the United States. So Houlton Intl Airport is the easternmost airport in America

In [13]:
dw = pd.read_csv("weather.csv")

## Displaying Weather Data

In [14]:
pd.read_csv("weather.csv")

origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
0        EWR  2013      1    1     1  39.02  26.06  59.37     270.0   
1        EWR  2013      1    1     2  39.02  26.96  61.63     250.0   
2        EWR  2013      1    1     3  39.02  28.04  64.43     240.0   
3        EWR  2013      1    1     4  39.92  28.04  62.21     250.0   
4        EWR  2013      1    1     5  39.02  28.04  64.43     260.0   
5        EWR  2013      1    1     6  37.94  28.04  67.21     240.0   
6        EWR  2013      1    1     7  39.02  28.04  64.43     240.0   
7        EWR  2013      1    1     8  39.92  28.04  62.21     250.0   
8        EWR  2013      1    1     9  39.92  28.04  62.21     260.0   
9        EWR  2013      1    1    10  41.00  28.04  59.65     260.0   
10       EWR  2013      1    1    11  41.00  26.96  57.06     260.0   
11       EWR  2013      1    1    13  39.20  28.40  69.67     330.0   
12       EWR  2013      1    1    14  39.02  24.08  54.68     280.0   
13       EWR  2013      1    1    15  37.94  24.08  57.04     290.0   
14       EWR  2013      1    1    16  37.04  19.94  49.62     300.0   
15       EWR  2013      1    1    17  35.96  19.04  49.83     330.0   
16       EWR  2013      1    1    18  33.98  15.08  45.43     310.0   
17       EWR  2013      1    1    19  33.08  12.92  42.84     320.0   
18       EWR  2013      1    1    20  32.00  15.08  49.19     310.0   
19       EWR  2013      1    1    21  30.02  12.92  48.48     320.0   
20       EWR  2013      1    1    22  28.94  12.02  48.69     320.0   
21       EWR  2013      1    1    23  28.04  10.94  48.15     310.0   
22       EWR  2013      1    2     0  26.96  10.94  50.34     310.0   
23       EWR  2013      1    2     1  26.06  10.94  52.25     330.0   
24       EWR  2013      1    2     2  24.98  10.94  54.65     330.0   
25       EWR  2013      1    2     3  24.08   8.96  51.93     320.0   
26       EWR  2013      1    2     4  24.08   8.96  51.93     330.0   
27       EWR  2013      1    2     5  24.08   8.96  51.93     330.0   
28       EWR  2013      1    2     6  24.08   8.96  51.93     310.0   
29       EWR  2013      1    2     7  24.98  10.04  52.50     300.0   
...      ...   ...    ...  ...   ...    ...    ...    ...       ...   
26085    LGA  2013     12   29    13  42.80  37.94  88.76      70.0   
26086    LGA  2013     12   29    14  41.00  37.94  93.19      60.0   
26087    LGA  2013     12   29    15  41.00  39.02  92.59      40.0   
26088    LGA  2013     12   29    16  41.00  37.94  88.76     350.0   
26089    LGA  2013     12   29    17  44.06  41.00  93.24     350.0   
26090    LGA  2013     12   29    18  42.08  39.02  88.81     330.0   
26091    LGA  2013     12   29    19  42.80  37.94  85.13     320.0   
26092    LGA  2013     12   29    20  42.08  37.94  86.89     320.0   
26093    LGA  2013     12   29    21  42.80  37.40  82.17     320.0   
26094    LGA  2013     12   29    22  42.98  37.04  79.38     300.0   
26095    LGA  2013     12   29    23  42.98  35.06  73.39     310.0   
26096    LGA  2013     12   30     0  42.08  33.98  72.78     320.0   
26097    LGA  2013     12   30     1  42.08  33.98  72.78     250.0   
26098    LGA  2013     12   30     2  41.00  33.98  75.88     240.0   
26099    LGA  2013     12   30     3  42.98  33.98  70.30     270.0   
26100    LGA  2013     12   30     4  41.00  33.08  73.19       0.0   
26101    LGA  2013     12   30     5  42.98  33.08  67.81     250.0   
26102    LGA  2013     12   30     6  42.98  33.98  70.30     230.0   
26103    LGA  2013     12   30     7  44.06  35.06  70.42     240.0   
26104    LGA  2013     12   30     8  44.06  33.98  67.45     260.0   
26105    LGA  2013     12   30     9  44.06  33.08  65.07     260.0   
26106    LGA  2013     12   30    10  42.98  33.80  70.28     330.0   
26107    LGA  2013     12   30    11  41.00  28.40  62.21     340.0   
26108    LGA  2013     12   30    12  37.94  23.00  54.51     330.0   
26109    LGA  2013     12   30

In [21]:
dw.loc[(dw['wind_speed'] >= 40)]

origin  year  month  day  hour   temp   dewp   humid  wind_dir  \
722      EWR  2013      1   31     4  60.80  59.00   93.79     230.0   
724      EWR  2013      1   31     6  57.20  53.60   87.74     270.0   
1009     EWR  2013      2   12     3  39.02  26.96   61.63     260.0   
9425     JFK  2013      1   31     4  53.60  53.06  100.00     200.0   
18131    LGA  2013      1   31     4  59.00  55.40   93.69     230.0   

       wind_speed  wind_gust  precip  pressure  visib             time_hour  
722      40.27730   58.68978    0.30       NaN    6.0  2013-01-31T09:00:00Z  
724      42.57886   55.23744    0.03       NaN   10.0  2013-01-31T11:00:00Z  
1009   1048.36058        NaN    0.00    1008.3   10.0  2013-02-12T08:00:00Z  
9425     42.57886   52.93588    0.20       NaN    2.0  2013-01-31T09:00:00Z  
18131    40.27730   55.23744    0.41       NaN    8.0  2013-01-31T09:00:00Z

### By filtering the wind_speed column EWR is the most windiest airport within the New York Area at 2013. With 1048 wind_speed.